### PR TITLE
Use XCom to correctly display the execution date from triggered DAG in TriggerDagOperator's extra link

### PR DESCRIPTION
**Pull request by o-nikolas**
Mitigate #19264 such that the extra link from the triggered dag at least loads the current date and
displays all current dag runs runs. See that issue for further discussion on future DB changes.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by o-nikolas** [comment]
@ljades @uranusjr Quick PR to remove the erroneous execution_date from the triggered url.

@uranusjr: Two of the tests are failing and I have no idea why. I can't reproduce it locally (those tests pass when I run them) and they aren't producing an exception or any logging, it just times out at 68% completion. Any ideas on how I can move forward here?

**Comment by uranusjr** [comment]
> it just times out at 68% completion. Any ideas on how I can move forward here?

Don't worry about those, the CI runner timed out.

**Comment by ashb** [comment]
This fix doesn't look right to me -- by not including the execution date it will link to the latest run instead.

**Comment by ljades** [comment]
> This fix doesn't look right to me -- by not including the execution date it will link to the latest run instead.

@ashb 
The issue working to be mitigated here is that the execution date parameter wasn't being used properly at all anyway and was linking to a poorly filtered view of the DAG, so mitigating it at least allows the user to view the latest and filter easily to the target.

**Comment by o-nikolas** [comment]
> > This fix doesn't look right to me -- by not including the execution date it will link to the latest run instead.
> 
> @ashb The issue working to be mitigated here is that the execution date parameter wasn't being used properly at all anyway and was linking to a poorly filtered view of the DAG, so mitigating it at least allows the user to view the latest and filter easily to the target.

Hey @ashb,
@ljades summarized it correctly. I know the original issue is quite long, but if you read that, it that will provide a lot of context so that this makes sense.

**Comment by ashb** [comment]
From that issue:

> So the webserver does not have access to the results of those computations when it's building the link

Store the triggered execution date in XCom and pull it in the operator link, like 

https://github.com/apache/airflow/blob/7622f5e08261afe5ab50a08a6ca0804af8c7c7fe/airflow/providers/google/cloud/operators/bigquery.py#L737

and use it like

https://github.com/apache/airflow/blob/7622f5e08261afe5ab50a08a6ca0804af8c7c7fe/airflow/providers/google/cloud/operators/bigquery.py#L63-L69

**Review by ashb — CHANGES_REQUESTED** [review]
Store date/runid (both please) in XCom and use to build the link.

**Comment by o-nikolas** [comment]
> Store date/runid (both please) in XCom and use to build the link.

Thanks for the feedback Ash! I'll play around with this and get back to this PR soon :+1:

**Comment by o-nikolas** [comment]
Hey @ashb, do you have a minute to take another look at this PR? Appreciate it!

File: airflow/operators/trigger_dagrun.py [hunk 1]
```diff
@@ -156,6 +166,11 @@ def execute(self, context: Dict):
             else:
                 raise e
 
+        # Store the execution date from the dag run (either created or found above) to
+        # be used when creating the extra link on the webserver.
+        ti = context['task_instance']
+        ti.xcom_push(key=XCOM_EXECUTION_DATE_ISO, value=dag_run.execution_date.isoformat())
```
Comment (ashb) [inline airflow/operators/trigger_dagrun.py:173]:
```suggestion
        ti.xcom_push(key=XCOM_EXECUTION_DATE_ISO, value=dag_run.execution_date.isoformat())
        ti.xcom_push(key="run_id", value=dag_run.run_id)
```

Too plese

Comment (o-nikolas) [reply airflow/operators/trigger_dagrun.py:173]: Hey @ashb, Sure I can make this change, but I'm curious, why do you want the run id pushed as well? Is there some other feature that depends on it? It isn't required for this feature as far as I can tell.

I'm probably missing something obvious XD

Comment (ashb) [reply airflow/operators/trigger_dagrun.py:173]: Not _right_ now, but I am slowly working towards making Airflow be able to have multiple runs for the _same_ date, so by adding this now it future proofs us and means we can change the URL without having to worry about backwards compat of "did we push run_id or not"

Comment (o-nikolas) [reply airflow/operators/trigger_dagrun.py:173]: Thanks for the context, it's helpful to add along with requested code changes! I'll push an update to the PR shortly.

**Review by uranusjr — APPROVED** [review]

**Comment by o-nikolas** [comment]
Hey @ashb your requested changes should be in. Do you have time to have a quick review of this one?

**Review by ashb — APPROVED** [review]
Sorry. LGTM!

**Pull request merged**